.github/workflows/check-release.yml

```diff
@@ -0,0 +1,81 @@
+name: Check Versions and Build-Publish
```

Comment:
Nitpicking but for consistency the name could be sentence case since all the other workflows' names are.